src/pages-home/ReservationRoom.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect } from 'react';
+// import PropTypes from 'prop-types';
+import {
+  Card,
+  CardBody,
+  CardFooter,
+  CardHeader,
+  Typography,
+} from '@material-tailwind/react';
+import { TrashIcon } from '@heroicons/react/24/outline';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import {
+  setMessageEmpty,
+  selectReservationMessage,
+  selectReservationStatus,
+  roomReservations,
+  deleteReservation,
+} from '../redux/Reservations/reservationsSlice';
+import useToken from '../redux/Auth/useToken';
+import { selectAuthenticatedUser } from '../redux/Auth/authSlice';
+import Loader from '../components/Loader';
+import ReservationDetail from './DetailsReservation';
+
+const Reservation = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const message = useSelector(selectReservationMessage);
+  const reservations = useSelector(roomReservations);
+  const status = useSelector(selectReservationStatus);
+  const currentUser = useSelector(selectAuthenticatedUser);
+  const isTokenSet = useToken();
+
+  const handleRemoveReservation = (reservationId) => {
+    const removeOptions = {
+      userId: currentUser.id,
+      reservationId,
+    };
+
+    dispatch(deleteReservation(removeOptions));
+  };
+
+  const checkAuthUser = () => {
+    if (!isTokenSet) navigate('/login');
+  };
+
+  const handleReservationMessage = () => {
+    if (message === 'Room has been successfully booked') dispatch(setMessageEmpty(''));
+  };
+
+  useEffect(() => {
+    checkAuthUser();
+    handleReservationMessage();
+  }, [message, isTokenSet]);
+
+  document.title = `Reservations: ${reservations.length}`;
+  return status === 'loading' ? (
+    <Loader />
+  ) : (
+    <div className="max-w-sm mt-3 mb-5 mx-auto flex flex-col gap-y-12 h-[95%]">
```

Comment:
- Kindly, it would be nice to remove the safe routing to look at your design rather than looking to render the reservation page.